connman/src/wispr.c

```diff
@@ -904,9 +936,35 @@ static bool wispr_portal_web_result(GWebResult *result, gpointer user_data)
 	return false;
 }
 
+static char *parse_proxy(const char *proxy)
+{
+	char *proxy_server;
+
+	if (!g_strcmp0(proxy, "DIRECT"))
+		return NULL;
+
+	if (!g_str_has_prefix(proxy, "PROXY "))
```

Comment:
Presumably proxy arg is always non-null?

Reply:
Part of upstream change. And yes, the calling function, [`proxy_callback()`](https://github.com/sailfishos/connman/pull/55/files#diff-b082b989c711bf47589cd0a7c33a91e9b550ba1ca13f17504efe6f13f22433b8R971) has check for `proxy` being NULL.